### DISCUSSION TITLE
CAD / OS Passwords doesn't work on Server editions

### DISCUSSION THREAD

**Discussion by c-michail** [rustdesk/rustdesk#8962] [Feature Request]
Hello,
  CAD / OS Password doesn't work on server edition. It doesn't initiates Ctrl-Alt-Del.
  I have to do "Insert Ctrl-Alt-Del" first to unlock and the click "OS Password".

  
Thank you

**Comment by rustdesk** [comment]
Yes, we need to execute Secure CAD from a system DLL instead from keyboard emulation.

**Reply by c-michail** [reply]
It's SendSAS from sas.dll it works on both client/server windows OS.

https://learn.microsoft.com/en-us/windows/win32/api/sas/nf-sas-sendsas

**Comment by c-michail** [comment]
Any news on this ? It's pretty simple!

**Comment by c-michail** [comment]
OS Passwords still doesn't work with Improve sas in 1.4.1 on Windows Server
